src/core/observer/index.js

```diff
@@ -188,6 +188,13 @@ export function defineReactive (
  * already exist.
  */
 export function set (obj: Array<any> | Object, key: any, val: any) {
+  if (arguments.length < 3) {
+    process.env.NODE_ENV !== 'production' && warn(
+      'Set now requires you to specifiy the object you are modifying' +
```

Comment:
Indeed there is. I'll need to re-commit because apparently amending doesn't work anymore.